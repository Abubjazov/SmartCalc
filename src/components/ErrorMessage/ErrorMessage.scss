@import '../../variables';

.error-message {
	margin: 113px 13px 0 13px;
	font-size: 18px;

	.error-message-p {
		text-align: center;
		color: $black;
		font-weight: 300;
		margin-bottom: 10px;
	}

	.error-message-a {
		display: block;
		margin-top: 13px;
		text-align: center;
		color: $deep-gray;
	}
}
