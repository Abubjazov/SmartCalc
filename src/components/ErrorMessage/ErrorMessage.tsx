import { NavLink } from 'react-router-dom'
import './ErrorMessage.scss'

export interface ErrorMessageProps {
	errorMessage: string | null
}

export const ErrorMessage = ({
	errorMessage,
}: ErrorMessageProps): JSX.Element => (
	<div role='alert' className='error-message'>
		<p className='error-message-p' tabIndex={0}>
			{errorMessage?.split('*')[0]}
		</p>
		<p className='error-message-p' tabIndex={0}>
			{errorMessage?.split('*')[1].split('>>')[0]}
		</p>

		<NavLink className='error-message-a' to='/smartcalc/'>
			Вернуться на главную
		</NavLink>
	</div>
)
