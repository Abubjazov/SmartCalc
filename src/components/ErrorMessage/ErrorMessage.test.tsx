import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

import { ErrorMessage } from './ErrorMessage'

describe('Component: ErrorMessage', () => {
	test('should render ErrorMessage', () => {
		const { asFragment } = render(
			<BrowserRouter>
				<ErrorMessage
					errorMessage={
						'Test Error String*Test Error String 2>>Test Error String 3'
					}
				/>
			</BrowserRouter>
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
