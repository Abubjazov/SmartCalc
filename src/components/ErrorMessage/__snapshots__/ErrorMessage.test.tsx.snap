// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ErrorMessage should render ErrorMessage 1`] = `
<DocumentFragment>
  <div
    class="error-message"
    role="alert"
  >
    <p
      class="error-message-p"
      tabindex="0"
    >
      Test Error String
    </p>
    <p
      class="error-message-p"
      tabindex="0"
    >
      Test Error String 2
    </p>
    <a
      class="error-message-a"
      href="/smartcalc/"
    >
      Вернуться на главную
    </a>
  </div>
</DocumentFragment>
`;
